### PR TITLE
Rename Module to Program

### PR DESCRIPTION
**Pull request by mariecwhite**

### CODE REVIEW

File: iree/jax/program_api.py [hunk 1]
```diff
@@ -416,69 +416,69 @@ def __new__(mcls,
...
 
-_STATIC_MODULE_ATTRIBUTES = (
+_STATIC_PROGRAM_ATTRIBUTES = (
     "_get_instance",
     "export_global",
     "get_class_info",
     "get_compiled_artifact",
     "get_info",
-    "get_mlir_module",
+    "get_mlir_program",
```
Comment (stellaraccident) [inline iree/jax/program_api.py:450]: So, in this case, this actually is returning an "miles module". Let's keep this one as is.

File: iree/jax/program_api.py [hunk 2]
```diff
@@ -600,17 +600,17 @@ def __setattr__(self, name, value):
 ################################################################################
 
 
-def _derive_module_export_name(class_name: str, explicit_name: Optional[str]):
-  """Returns an appropriate module export name given a class name and override.
+def _derive_program_export_name(class_name: str, explicit_name: Optional[str]):
```
Comment (stellaraccident) [inline iree/jax/program_api.py:603]: Ditto here: this is an mlir module name

File: tests/legacy_xla/frontend_test.py [hunk 3]
```diff
@@ -22,7 +22,7 @@
 import numpy as np
 
 # pytype thinks iree.jax is jax.
-# pytype: disable=module-attr
```
Comment (stellaraccident) [inline tests/legacy_xla/frontend_test.py:25]: Copy pasta?

**Review by stellaraccident — APPROVED** [review]
Thanks - just a few nits

File: iree/jax/frontend.py [hunk 4]
```diff
@@ -25,12 +25,12 @@
...
   import jax
-except ModuleNotFoundError as e:
-  raise ModuleNotFoundError("iree.jax requires 'jax' and 'jaxlib' to be "
+except ProgramNotFoundError as e:
+  raise ProgramNotFoundError("iree.jax requires 'jax' and 'jaxlib' to be "
                             "installed in your python environment.") from e
 
 # pytype thinks iree.jax is jax.
-# pytype: disable=module-attr
+# pytype: disable=program-attr
```
Comment (phoenix-meadowlark) [inline iree/jax/frontend.py:33]: I think these should be left as is.

**Review by phoenix-meadowlark — APPROVED** [review]

File: tests/legacy_xla/frontend_test.py (hunk 3 above)
Comment (mariecwhite) [reply tests/legacy_xla/frontend_test.py]: Ooops

File: iree/jax/program_api.py (hunk 2 above)
Comment (mariecwhite) [reply iree/jax/program_api.py]: Renamed to module.

File: iree/jax/program_api.py (hunk 1 above)
Comment (mariecwhite) [reply iree/jax/program_api.py]: Reverted

File: iree/jax/frontend.py (hunk 4 above)
Comment (mariecwhite) [reply iree/jax/frontend.py]: Thanks, changed back to use Module / module

**Pull request merged**